Content.Server/GameTicking/Rules/TraitorRuleSystem.cs

```diff
@@ -129,20 +139,50 @@ public bool MakeTraitor(EntityUid traitor, TraitorRuleComponent component, bool
         return true;
     }
 
+    private (Note[]?, string) RequestUplink(EntityUid traitor, FixedPoint2 startingBalance, string briefing)
+    {
+        var pda = _uplink.FindUplinkTarget(traitor);
+        Note[]? code = null;
+
+        var uplinked = _uplink.AddUplink(traitor, startingBalance, giveDiscounts: true);
+
+        if (pda is not null && uplinked)
+        {
+            // Codes are only generated if the uplink is a PDA
+            code = EnsureComp<RingerUplinkComponent>(pda.Value).Code;
+
+            // If giveUplink is false the uplink code part is omitted
+            briefing = string.Format("{0}\n{1}",
+                briefing,
+                Loc.GetString("traitor-role-uplink-code-short", ("code", string.Join("-", code).Replace("sharp", "#"))));
+            return (code, briefing);
+        }
+        else if (pda is null && uplinked)
+        {
+            briefing = string.Format("{0}\n{1}", briefing, Loc.GetString("traitor-role-uplink-implant-short"));
```

Comment:
```suggestion
            briefing += "\n" + Loc.GetString("traitor-role-uplink-implant-short");
```
would be a little cleaner imo